File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/cbaps/pages/portal_home_page.py
"""
PortalHomePage: Shared entry point for CBAPS and DAWMS applications
User selects which application to access from portal
"""

import logging
from playwright.sync_api import Page, Locator
from cbaps.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class PortalHomePage:
    """
    PortalHomePage represents the portal landing page.
    This is the Python equivalent of the Java PortalHomePage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize PortalHomePage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.cbaps_link: Locator = page.locator("a:has-text('CBAPS')")
        self.dawms_link: Locator = page.locator("a:has-text('DAWMS')")
        
        # Stability anchor: wait for portal to load
        pwm.wait_visible("text=Application Portal")
        logger.info("Portal Home Page loaded successfully.")
    
    def navigate_to_portal(self, portal_url: str) -> 'PortalHomePage':
        """
        Navigate to portal homepage
        
        Args:
            portal_url: URL of the portal
            
        Returns:
            self for method chaining
        """
        self.page.goto(portal_url)
        self.pwm.wait_visible("text=Application Portal")
        logger.info(f"Navigated to Portal: {portal_url}")
        return self
    
    def open_cbaps(self) -> 'CBAPSDashboardPage':
        """
        Open CBAPS application from portal
        
        Returns:
            CBAPSDashboardPage instance
        """
        from cbaps.pages.cbaps_dashboard_page import CBAPSDashboardPage
        
        self.pwm.click(self.cbaps_link)
        logger.info("Clicked CBAPS link from portal.")
        return CBAPSDashboardPage(self.page, self.pwm)
    
    def get_title(self) -> str:
        """Get portal page title"""
        return self.page.title()

File: 01_projects/code/java-playwright/cbaps-dawms/basic-pom/src/test/java/com/cbaps/pages/PortalHomePage.java
package com.cbaps.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.cbaps.library.PlaywrightManager;

/**
 * PortalHomePage: Shared entry point for CBAPS and DAWMS applications
 * User selects which application to access from portal
 */
public class PortalHomePage {

	private static final Logger log = LoggerFactory.getLogger(PortalHomePage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator cbapsLink;
	private final Locator dawmsLink;

	public PortalHomePage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Locators for application links
		this.cbapsLink = page.locator("a:has-text('CBAPS')");
		this.dawmsLink = page.locator("a:has-text('DAWMS')");

		// Stability anchor: wait for portal to load
		pwm.waitVisible("text=Application Portal");
		log.info("Portal Home Page loaded successfully.");
	}

	/**
	 * Navigate to portal homepage
	 */
	public PortalHomePage navigateToPortal(String portalUrl) {
		page.navigate(portalUrl);
		pwm.waitVisible("text=Application Portal");
		log.info("Navigated to Portal: " + portalUrl);
		return this;
	}

	/**
	 * Open CBAPS application from portal
	 * @return CBAPSDashboardPage
	 */
	public CBAPSDashboardPage openCBAPS() {
		pwm.click(cbapsLink);
		log.info("Clicked CBAPS link from portal.");
		return new CBAPSDashboardPage(page, pwm);
	}

	/**
	 * Get portal page title
	 */
	public String getTitle() {
		return page.title();
	}
}
